xrpl/asyncio/wallet/wallet_generation.py

```diff
@@ -50,6 +55,14 @@ async def generate_faucet_wallet(
     .. # noqa: DAR402 exception raised in private method
     """
     faucet_url = get_faucet_url(client.url, faucet_host)
+
+    if faucet_url == _HOOKS_V2_TEST_FAUCET_URL and wallet is not None:
```

Comment:
iirc this is false (according to Discord conversations) - we should add support for this